src/guiFormSpecMenu.cpp

```diff
@@ -2388,9 +2389,15 @@ void GUIFormSpecMenu::drawList(const ListDrawSpec &s, int phase,
 						utf8_to_wide(item.getDefinition(m_client->idef()).description);
 				else
 					tooltip_text = utf8_to_wide(desc);
-				// Show itemstring as fallback for easier debugging
-				if (!item.name.empty() && tooltip_text.empty())
-					tooltip_text = utf8_to_wide(item.name);
+
+				if (!item.name.empty()) {
+					//Append itemstring if enabled in settings
+					if (tooltip_text.empty())
+						tooltip_text = utf8_to_wide(item.name);
+					if (m_tooltip_show_itemstring){
```

Comment:
`{` and counterpart bracket can be removed, otherwise add a space between `){`, see [code style guidelines](http://dev.minetest.net/Code_style_guidelines).